src/iplant_groups/clients/grouper.clj

```diff
@@ -591,10 +592,12 @@
 
 (defn update-group-privileges
   [username replace? group-name subject-ids privilege-names]
-  (when replace?
-    (remove-group-privileges username group-name subject-ids all-group-privileges))
-  (when (seq privilege-names)
-    (add-group-privileges username group-name subject-ids privilege-names)))
+  (let [privs-to-add    (set privilege-names)
+        privs-to-remove (set/difference all-group-privileges privs-to-add)]
+    (when (seq privs-to-add)
+      (add-group-privileges username group-name subject-ids privs-to-add))
+    (when (and replace? (seq privs-to-remove))
+      (remove-group-privileges username group-name subject-ids privs-to-remove))))
```

Comment:
I changed the order of operations here so that the new privileges are added before the old ones are removed.